### PR TITLE
Automator update envoy SHA to istio/proxy + Add AUTOMATOR_SHA_COMMIT_DATE

### PR DESCRIPTION
**Pull request by carolynhu**
https://github.com/istio/test-infra/issues/2937

### CODE REVIEW

File: prow/config/jobs/envoy.yaml [hunk 1]
```diff
@@ -33,6 +33,22 @@ jobs:
...
+  type: postsubmit
+  command:
+  - ../test-infra/tools/automator/automator.sh
+  - --org=istio
+  - --repo=proxy
+  - "--title=Automator: update envoy@$AUTOMATOR_SRC_BRANCH in $AUTOMATOR_ORG/$AUTOMATOR_REPO@$AUTOMATOR_BRANCH"
+  - --labels=auto-merge,release-notes-none
+  - --modifier=update_envoy_dep
+  - --token-path=/etc/github-token/oauth
+  - --cmd=bin/update_envoy.sh $AUTOMATOR_SHA
```
Comment (howardjohn) [inline prow/config/jobs/envoy.yaml]: Will you be adding this?

Comment (carolynhu) [reply prow/config/jobs/envoy.yaml]: yes! working on that

File: prow/config/jobs/envoy.yaml [hunk 2]
```diff
@@ -33,6 +33,23 @@ jobs:
...
 
+periodics:
+- interval: 168h
+  name: update-proxy
+  command:
+  - ../test-infra/tools/automator/automator.sh
+  - --org=istio
+  - --repo=proxy
+  - "--title=Automator: update envoy@$AUTOMATOR_SRC_BRANCH in $AUTOMATOR_ORG/$AUTOMATOR_REPO@$AUTOMATOR_BRANCH"
+  - --labels=auto-merge,release-notes-none
```
Comment (bianpengyuan) [inline prow/config/jobs/envoy.yaml]: nit: proxy repo does not need release notes label.

File: prow/config/jobs/envoy.yaml [hunk 3]
```diff
@@ -33,6 +33,23 @@ jobs:
...
+  - --modifier=update_envoy_dep
```
Comment (bianpengyuan) [inline prow/config/jobs/envoy.yaml:45]: Not familiar with the configuration here, but is `update_envoy_dep` a keyword that points to something? or just a random name?

**Review by bianpengyuan — COMMENTED** [review]
How do we make sure this job only runs on master?

File: prow/config/jobs/envoy.yaml (hunk 3 above)
Comment (carolynhu) [reply prow/config/jobs/envoy.yaml:45]: https://github.com/istio/test-infra/tree/master/tools/automator `A segment of text to append to the branch name of the fork for additional context. This option is optional and will default to automator.sh.`

File: prow/config/jobs/envoy.yaml [hunk 4]
```diff
@@ -33,6 +33,23 @@ jobs:
     value: "/home/prow/go/src/istio.io/envoy"
   command: [./ci/do_ci.sh, bazel.release]
 
+periodics:
+- interval: 168h
```
Comment (bianpengyuan) [inline prow/config/jobs/envoy.yaml]: nit: do we support something like `1w` or `7d`?

**Review by bianpengyuan — APPROVED** [review]

**Comment by carolynhu** [comment]
put on hold: need to verify this periodic job config

File: prow/config/jobs/envoy.yaml (hunk 4 above)
Comment (carolynhu) [reply prow/config/jobs/envoy.yaml]: sorry, it's golang parseduration impl, in which `Valid time units are "ns", "us" (or "µs"), "ms", "s", "m", "h".`

**Review by howardjohn — APPROVED** [review]

**Pull request merged**